tests/generation/test_utils.py

```diff
@@ -2455,220 +2037,6 @@ def test_diverse_beam_search(self):
             ],
         )
 
-    def test_max_length_backward_compat_greedy(self):
```

Comment:
We should still have at least one `test_max_length_backward_compat` test which calls `model.generate`

Reply:
I removed this tests, because we can only check it if the (now private) generate methods are called directly. When calling from the `model.generate`, the `max_length` get defined by default from the generation config.

We have a test `test_default_max_length_warning`, which check max_length warnings when calling directly `generate`